File: ios/ExpoAutoSmsVerificationModule.swift
import ExpoModulesCore

public class ExpoAutoSmsVerificationModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAutoSmsVerification")

    Events("onSmsReceived", "onSmsTimeout", "onSmsError")

    AsyncFunction("startSmsRetriever") { () throws in
      throw UnavailabilityError()
    }

    AsyncFunction("stopSmsRetriever") { () throws in
      throw UnavailabilityError()
    }

    AsyncFunction("getMessageHash") { () throws -> [String] in
      throw UnavailabilityError()
    }
  }
}

File: src/ExpoAutoSmsVerificationModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { ExpoAutoSmsVerificationModuleEvents } from './ExpoAutoSmsVerification.types';

declare class ExpoAutoSmsVerificationModule extends NativeModule<ExpoAutoSmsVerificationModuleEvents> {
  /**
   * Starts the Android SMS Retriever API and registers a BroadcastReceiver.
   * The listener stays active for up to 5 minutes.
   * Fires `onSmsReceived`, `onSmsTimeout`, or `onSmsError` events.
   */
  startSmsRetriever(): Promise<void>;

  /**
   * Manually stops the SMS Retriever before the 5-minute timeout.
   */
  stopSmsRetriever(): Promise<void>;
}

export default requireNativeModule<ExpoAutoSmsVerificationModule>('ExpoAutoSmsVerification');
